benchmark/distinct_user_id/nlohmann_json_sax.h

```diff
@@ -17,9 +17,11 @@ struct nlohmann_json_sax {
         Handler(std::vector<uint64_t> &r) : result(r) { }
 
         bool key(string_t& val) override {
-            // Assume that valid user/id pairs are only in user objects
-            if (val.compare("user") == 0) { user = true; }
-            else if (user && val.compare("id") == 0) { user_id = true; }
+            // Assume that valid user/id pairs appear only once in main array of user objects
+            if (user) { // If already found user object, find id key
+                if (val.compare("id") == 0) { user_id = true; }
+            }
+            else if (val.compare("user") == 0) { user = true; } // Otherwise, find user object
```

Comment:
I think this should do one string comparison per key. However, for some reason, performance has decreased to 0.66 GB/s. I was not able to pinpoint the reason why.

Reply:
For some reasons, when I compile this without nlohmann_sax, rapidjson_sax has a similar performance as before (0.71GB/s), but when I compile with nlohmann_sax included, the performance of rapidjson_sax drops at 0.66GB/s.